design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+
+Case 2 - The user changes the selected formula.
+Once the user selects a different formula parameter values are not retained and they see empty parameters and an empty column again. To see the value they will have to add compatible params again.
+
+### Scenario 6 - Click outside the inspector
+
+#### Scenario 6.a - Clicks outside before adding name or selecting formula
+
+If the users click outside the inspector before even touching the formula settings that is adding name or selecting formula, there won't be any empty column added. And there would be absolutely no change in the explorer.
+
+#### Scenario 6.b - Clicks outside after adding name or selecting formula
+
+If there is any activity in the formula settings that is if the name of the column is added or the formula is selected, an empty column will be added. 
+
+Once users click outside, they will see a warning on the column so users know there is something left to do with the formula column. 
+
+![](https://share.balsamiq.com/c/79ZrfbdaMgz8rfuHN3bRDY.png)
+
+### Scenario 7 - Quit the explorer
+
+Case 1 - The preview column is empty
+If the preview column is empty, there won't be any save changes warning modal. Since there are no significant changes and there is nothing to be saved. The modal would only have a warning about whether to close the explorer or not.
+![](https://share.balsamiq.com/c/ffSKeCLi8D3DXfkRN8qrV1.png)
+
+Case 2 - The preview column is filled
+If the preview column is filled but not yet added to the query, the user will be asked in the modal if they want it to save it to the query directly or just close the explorer without saving anything.
+![](https://share.balsamiq.com/c/79ZrfbdaMgz8rfuHN3bRDY.png)
+
+
+## Interactions
+
+### Modal for warning before qutting the explorer
+
+The modals are for warning the users about the consequences of closing the data explorer. There is no action for saving query in the data explorer view. The modal provides the action of saving the column if there are significant changes that is there a filled preview column. If there are no significant changes there is only warning of closing the data explorer.
+
+![](https://share.balsamiq.com/c/5Y3TjZ551mZdf5Q4cpBCNi.png)
+
+### Column preview while adding the formula
+
+The column preview is necessary for the user to see the reflected changes when they interact with the formula settings. The different states of the column preview can be seen below. 
+
+1. Active empty column preview (Green highlight) - Users will see the active empty column preview when they are interacting with the formula settings in the inspector.
+
+2. Inactive empty column with a warning (Yellow highlight) - Users will see the inactive column preview with a warning when the user has click outside the inspector. 
+
+3. Erroneous empty column (Red highlight) - Users will see erroneous column when there are errors in formula settings inputs.
+
+![](https://share.balsamiq.com/c/bhzoPkNR4MtQnpb4mB2hws.png)
+
```

Comment:
I thought we'd moved past the need for the preview? Because we don't have a save action, it won't be obvious when the preview ends and the new column is added, which could result in users losing their progress inadvertently.

Reply:
yep.